lightning/src/ln/channelmanager.rs

```diff
@@ -8787,25 +8794,32 @@ where
 
 		let mut pending_offers_messages = self.pending_offers_messages.lock().unwrap();
 		if !offer.paths().is_empty() {
-			// Send as many invoice requests as there are paths in the offer (with an upper bound).
-			// Using only one path could result in a failure if the path no longer exists. But only
-			// one invoice for a given payment id will be paid, even if more than one is received.
+			// Send as many invoice requests as there are paths in the offer using as many different
+			// reply_paths possible (with an upper bound).
+			// Using only one path could result in a failure if the path no longer exists.
+			// But only one invoice for a given payment ID will be paid, even if more than one is received.
 			const REQUEST_LIMIT: usize = 10;
-			for path in offer.paths().into_iter().take(REQUEST_LIMIT) {
+			reply_paths
+				.iter()
+				.flat_map(|reply_path| offer.paths().iter().map(|path| (path.clone(), reply_path.clone())))
+				.take(REQUEST_LIMIT)
+				.for_each(|(path, reply_path)| {
+					let message = new_pending_onion_message(
+						OffersMessage::InvoiceRequest(invoice_request.clone()),
+						Destination::BlindedPath(path.clone()),
+						Some(reply_path.clone()),
+					);
+					pending_offers_messages.push(message);
+				});
```

Comment:
Should be able to reduce the number of clones here. Use `reply_paths.into_iter()` and remove any later clones when no longer necessary. Might need to use a loop instead of `for_each`.

Reply:
Done! Utilized `move` here, to remove the clonings in `flat_map()`
Addressed in [4032289](https://github.com/lightningdevkit/rust-lightning/commit/40322898323c2656779c1ff23e54310229ecd671)
Thanks!